pages/[user]/[type].tsx

```diff
@@ -61,9 +65,12 @@ export default function Type(props) {
         var i = props.user.startTime;
     }
     
+    // Adding first availability time
+    times.push(dayjs(selectedDate).tz(props.user.timeZone).hour(Math.floor(i / 60)).minute(i % 60).startOf(props.eventType.length, 'minute'));
```

Comment:
Say I am in UTC+4, and the agent I'm booking with is in UTC+1 - my browser timezone will be ignored and the timeslots will be shown using the UTC+1 timeZone as configured by the agent.

Reply:
This line fixes a bug that was there before adding timezone support. 

Let's ignore for a second the timezones. I set the start of my day at 10 am and the end at 7 pm. Without this line, the calendar shows the first available slot at 11 am, missing the actual start of my day.

Does it make sense to you?

Reply:
Talking about your example, we need to add the start of the day to make a good example.

If I am in UTC+4 the agent I'm booking with is in UTC-1 and the agent sets the start of the day at 10 am (UTC-1), I will see the first available option at the same time, but displayed in my time zone, UTC+4, so 3 pm.

The thing I maybe mistakenly took for granted is that these are online events, making it reasonable to show them in the time of the viewer.

The mechanism is different for an in-person event, as the time zone should be locked to the location of the event. Calendly implements this.

Reply:
I am debugging this line and it causes a bug. Turns out that when you have zero events in your calendar `busy.forEach` will run on an empty array.

Moving this
```js
// Check if time has passed
if (dayjs(times[i]).isBefore(dayjs())) {
  times.splice(i, 1);
}
```
outside of the forEach loop does the job.